<launch>
    <!-- Detection configuration -->
    <arg name="weights"                 default="$(find yolov5_ros)/src/yolov5/best.pt"/>
    <arg name="data"                    default="$(find yolov5_ros)/src/yolov5/data/data.yaml"/>
    <arg name="confidence_threshold"    default="0.85"/>
    <arg name="iou_threshold"           default="0.5"/>
    <arg name="maximum_detections"      default="1000"/>
    <arg name="device"                  default="cpu"/>
    <arg name="agnostic_nms"            default="true"/>
    <arg name="line_thickness"          default="3"/>
    <arg name="dnn"                     default="true"/>
    <arg name="half"                    default="false"/>

    <!-- replace imgsz -->
    <arg name="inference_size_h"        default="640"/>
    <arg name="inference_size_w"        default="640"/>

    <!-- Visualize using OpenCV window -->
    <arg name="view_image"              default="false"/>

    <!-- ROS topics -->
    <arg name="input_image_topic"       default="/camera_front/color/image_raw"/>
    <arg name="output_topic"            default="/yolov5/detections"/>

    <!--find depth topic-->
    <arg name="depth_image_topic"       default="/camera_front/depth/image_rect_raw"/>
    <arg name="prediction"              default="/yolov5/pred_img"/>

    <!--find controller topic-->
    <arg name="controller_topic"        default="/anicar/vesc/commands/motor/unsmoothed_speed"/>

    <!-- Optional topic (publishing annotated image) -->
    <arg name="publish_image"           default="ture"/>


    <node pkg="yolov5_ros" name="detect" type="detect.py" output="screen">
        <param name="weights"                   value="$(arg weights)"/>
        <param name="data"                      value="$(arg data)"/>
        <param name="confidence_threshold"      value="$(arg confidence_threshold)"/>
        <param name="iou_threshold"             value="$(arg iou_threshold)" />
        <param name="maximum_detections"        value="$(arg maximum_detections)"/>
        <param name="device"                    value="$(arg device)" />
        <param name="agnostic_nms"              value="$(arg agnostic_nms)" />
        <param name="line_thickness"            value="$(arg line_thickness)"/>
        <param name="dnn"                       value="$(arg dnn)"/>
        <param name="half"                      value="$(arg half)"/>

        <param name="inference_size_h"          value="$(arg inference_size_h)"/>
        <param name="inference_size_w"          value="$(arg inference_size_w)"/>

        <param name="input_image_topic"         value="$(arg input_image_topic)"/>
        <param name="output_topic"              value="$(arg output_topic)"/>

        <param name="depth_image_topic"         value="$(arg depth_image_topic)"/>
        <param name="prediction"                value="$(arg prediction)"/>

        <param name="color_camera_info_topic"   value="/camera_front/color/camera_info"/>
        <param name="depth_camera_info_topic"   value="/camera_front/depth/camera_info"/>

        <param name="controller_topic"          value="$(arg controller_topic)"/>

        <param name="view_image"                value="$(arg view_image)"/>

        <param name="publish_image"             value="$(arg publish_image)"/>

    </node>


</launch>